src/nvim/ex_getln.c

```diff
@@ -1669,7 +1672,16 @@ getcmdline (
     int indent               // indent for inside conditionals
 )
 {
-  return command_line_enter(firstc, count, indent);
+  if (ui_is_external(kUICmdline)) {
+    Array args = ARRAY_DICT_INIT;
+    ui_event("cmdline_enter", args);
```

Comment:
A separate ```cmdline_enter``` event can be helpful in this scenario:

```:<C-R>=1+2<CR><CR>```

The events would be ```cmdline_enter``` ```cmdline_enter``` ```cmdline_leave``` ```cmdline_leave```  

If we only have cmdline_show, the events will be:

```cmdline_show``` ```cmdline_show``` ```cmdline_leave```

You wouldn't know when you entered the "sub" cmdline.